compiler/runtime.go

```diff
@@ -137,32 +146,37 @@ func (e *stringExpressionEvaluator) Type() semantic.MonoType {
 	return semantic.BasicString
 }
 
+// TODO
 func (e *stringExpressionEvaluator) Eval(ctx context.Context, scope Scope) (values.Value, error) {
 	var b strings.Builder
 	for _, p := range e.parts {
 		v, err := p.Eval(ctx, scope)
 		if err != nil {
 			return nil, err
 		}
+		defer v.Release()
 
 		if v.IsNull() {
 			return nil, errors.New(codes.Invalid, "string expression evaluated to null")
 		}
 		b.WriteString(v.Str())
 	}
-	return values.NewString(b.String()), nil
+	s := values.NewString(b.String())
```

Comment:
`values.NewString` should return a value with a reference count of one so I think this retain is incorrect. Although, in reality, the retain is a no-op and does nothing.